contracts/drafts/issuance/Issuance.sol

```diff
@@ -113,6 +116,34 @@ contract Issuance is Ownable, StateMachine {
         emit InvestmentAdded(msg.sender, _amount);
     }
 
+    function withdraw() external nonReentrant {
+        require(currentState == "DISTRIBUTING", "Cannot withdraw now.");
+        require(investments[msg.sender] > 0, "No investments found.");
+        if (nextInvestor < investors.length) {
```

Comment:
You probably don't need `nextInvestor` anymore. Investors can come and serve themselves in any order.